README.md

```diff
@@ -24,6 +27,84 @@ The Xcode project is configured to use 4 spaces for indentation.
 For linting Swift source code, we use [SwiftLint](https://github.com/realm/SwiftLint).
 A configuration for [SwiftFormat](http://github.com/nicklockwood/SwiftFormat) is also included.
 
+## App architecture
+
+### Core principles
+
+This app is built using SwiftUI and targets iOS 15 and higher. We use SwiftUI as much as possible, but fall back to UIKit views using view(controller) representables where needed.
+
+We try to stick to the Apple conventions and write idiomatic SwiftUI code. Do things the Apple way. Lean in to the platform instead of fighting it.
+
+Keep it simple. Less is more.
+
+### Architecture patterns
+
+We use the [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) pattern, combined with dependency injection.
+
+#### Clean Architecture layers
+
+- _UI_ with SwiftUI
+- _Presentation_ with the ViewModel
+- _Domain_ for domain models, UseCases and other domain logic.
+- _Data_ for data storage and retrieval.
+
+#### Use cases
+
+- Use cases are single-purpose: GetUserUseCase, but also: GetUserWithArticlesUseCase.
+- Use cases can call other use-cases.
+- Use cases do not have state, state preferably lives in the data layer.
+
+#### Dependency injection
+
+Dependency injection (DI) means that objects don't instantiate the objects or configurations that they require themselves, but they are passed in from the outside.
+This is useful, because it makes code easier to test and easier to change. It promotes good separation of concerns.
+
+We use [Factory](https://github.com/hmlongco/Factory) as a DI container.
```

Comment:
I would also mention [swift-dependencies](https://github.com/pointfreeco/swift-dependencies) as an option, because 
1) I'm prejudiced :trollface:  
2) but it also makes transition to TCA slightly easier if an app decides to move to TCA.

Fortunately, both libraries are conceptually very similar so it's easy to reason about one or the other :relaxed: